src/test/java/com/marek/onlinebookstore/repository/book/BookRepositoryTest.java

```diff
@@ -0,0 +1,39 @@
+package com.marek.onlinebookstore.repository.book;
+
+import com.marek.onlinebookstore.model.Book;
+import java.util.List;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.test.context.jdbc.Sql;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+class BookRepositoryTest {
+
+    @Autowired
+    private BookRepository bookRepository;
+
+    @Test
+    @DisplayName("""
+            Find all books by category id
+            """)
+    @Sql(scripts = {"classpath:db/books/repository/Add-book-to-books-table.sql"},
```

Comment:
SQL scripts for setting up and tearing down the database for testing are hard-coded in the test. This can lead to problems when the database schema changes, as the scripts will also need to be updated.